Two new options: MythTV PVR Recordings and Secondary Title Lookup

ineffablebob opened the pull request:
Added two new options to the Scrobbler function. Both show up in the
configuration under Scrobbler and default to false.

MythTV PVR recordings: attempt to scrobble recordings from the MythTV
PVR add-on, based on the show title/episode name. See comments in
service.py (onPlayBackStarted function) for more info.

Secondary Title Lookup: If scrobbler can't find a show, try the lookup
again using Trakt text search, which works with secondary show titles.
See comments in scrobbler.py (__scrobble function) for more info.

Razzeee commented on README.md:
```diff
@@ -65,9 +65,7 @@ or
-8. Watch *something* and see it show up on Trakt.tv!
-
-Please note that *something* does not cover all Kodi possible streaming sources. Local files and strm files scrapped to your library should be OK, however generic third party streaming addons can fail. It is up to the developers of these addons to be supported by this plugin. Please take a look https://github.com/trakt/script.trakt/wiki/Providing-id's-to-facilitate-scrobbling
+8. Watch something and see it show up on Trakt.tv!
```
Please don't remove this

ineffablebob replied:
Whoops, merge error, didn't intend to change README at all. Will revert.

Razzeee commented on service.py:
```diff
@@ -9,7 +9,9 @@
 import json
+import re
 import AddonSignals
+import re
```
I don't think we need to import this twice :)

Razzeee commented on service.py:
```diff
@@ -445,32 +451,33 @@ def onPlayBackStarted(self):
                     if self.type == 'episode':
                         logger.debug("[traktPlayer] onPlayBackStarted() - Doing multi-part episode check.")
-                        result = kodiUtilities.kodiJsonRequest({'jsonrpc': '2.0', 'method': 'VideoLibrary.GetEpisodeDetails', 'params': {'episodeid': self.id, 'properties': ['tvshowid', 'season', 'episode', 'file']}, 'id': 1})
+                        result = utilities.kodiJsonRequest({'jsonrpc': '2.0', 'method': 'VideoLibrary.GetEpisodeDetails', 'params': {'episodeid': self.id, 'properties': ['tvshowid', 'season', 'episode', 'file']}, 'id': 1})
```
This will likely fail, as the method does not exist in that class

Razzeee commented on service.py:
```diff
@@ -485,25 +492,175 @@ def onPlayBackStarted(self):
-
-                # send dispatch
-                self.action(data)
+            else:
```
Merge error?

Razzeee commented on traktapi.py:
```diff
@@ -332,6 +332,10 @@ def getShowSummary(self, showId):
             return Trakt['shows'].get(showId)
 
+    def getShowWithAllEpisodesList(self, showId):
+        with Trakt.configuration.http(retry=True):
```
You might want to add a timeout here, as this is a read call `(retry=True, timeout=90):`

Razzeee commented on service.py:
```diff
@@ -445,32 +451,33 @@ def onPlayBackStarted(self):
                         logger.debug("[traktPlayer] onPlayBackStarted() - Playing a non-library 'movie' - %s (%s)." % (data['title'], data.get('year', 'NaN')))
                     elif showtitle:
-                        title, season, episode = utilities.regex_tvshow(showtitle)
+                        title, season, episode = utilities.regex_tvshow(False, showtitle)
```
This and the following lines seem like an error, or there is more code missing

Razzeee commented:
You might want to also update the changelog.

Maybe split the two features commit wise, if your able to do that.

One thing I'm not to keen on, is that it might produce quiet some log spam.

ineffablebob commented:
Thanks, I'll go back through all your comments and make updates. I think I can probably separate the two features into separate commits.

As for the log spam, none of it should show up unless the user has debug logging enabled anyway, so I didn't think it was a serious problem. I've found the messages very useful when problem occur.

ineffablebob commented:
I believe I've addressed the comments from earlier. I attempted to separate the changes by putting them on two different branches, then merging both into my master branch (which were then automatically added to this pull request). I'm not sure whether that's exactly what you're looking for, but hopefully it's clear enough.

Razzeee commented on traktapi.py:
```diff
@@ -343,6 +347,13 @@ def getIdLookup(self, id, id_type):
             return result
 
+    def getTextQuery(self, query, type, year):
+        with Trakt.configuration.http(retry=True):
```
with Trakt.configuration.http(retry=True, timeout=90)

ineffablebob replied:
Sorry, my bad. I added that in the getShowWithAllEpisodesList function but didn't copy it here. Fixed.

Razzeee commented:
Looks good, do you know how to rebase/squash? Which git client are you using?

Razzeee commented on service.py:
```diff
@@ -410,6 +412,9 @@ def onPlayBackStarted(self):
 
                 data = {'action': 'started'}
 
+                if (kodiUtilities.getSettingAsBool('scrobble_mythtv_pvr')):
```
This seems to be far away from the actual mythtv code?

ineffablebob replied:
This one is just a debug message - if you don't put it this far up in the branch tree, it's impossible to tell if a failure to scrobble is due to incorrect incoming data, or because the option is disabled by the user.

ineffablebob commented:
I'm slowly but surely learning about rebase! I now have just the two commits, first the MythTV feature, and then the secondary title lookup added in. These include the extra timeout you commented on earlier today. (Technically there's the original combined code plus a revert in there too, but I don't see how to get rid of those since they're the first commit on the branch.) For some reason I had to force push from my local to github - the regular push kept failing - but it seems OK now.

Razzeee commented:
You have to force push, if you change history. Which is rebasing/squashing :)

ineffablebob commented:
Good to know, thanks. Let me know if there's anything else I should change!

Razzeee commented:
Well can you do that? Maybe make a backup before just in case. (just copy the local dir off to a different location)

ineffablebob commented:
OK, all set. I dropped the original+revert commits, so now you've got just the 2: one for the MythTV feature, then another to add the secondary title lookup feature.

Razzeee commented:
Thank you, I'm gone for vacation this week, so I will probably check back in 10 days or so

Razzeee commented:
Sorry for taking my time and thanks for the contribution.
